apps/web/src/views/Swap/V3Swap/containers/SwapCommitButtonV1.tsx

```diff
@@ -0,0 +1,391 @@
+  const relevantTokenBalances = useCurrencyBalances(account ?? undefined, [
+    inputCurrency ?? undefined,
+    outputCurrency ?? undefined,
+  ])
+  const currencyBalances = {
+    [Field.INPUT]: relevantTokenBalances[0],
+    [Field.OUTPUT]: relevantTokenBalances[1],
+  }
+
+  // check whether the user has approved the router on the input token
+  const { approvalState, approveCallback, revokeCallback, currentAllowance, isPendingError } = useApproveCallback(
+    amountToApprove,
+    routerAddress,
+  )
+  const { priceImpactWithoutFee } = useMemo(
+    () => (!showWrap ? computeTradePriceBreakdown(trade) : {}),
+    [showWrap, trade],
+  )
+  const swapInputError = useSwapInputError(trade, currencyBalances)
+  const parsedAmounts = useParsedAmounts(trade, currencyBalances, showWrap)
+  const parsedIndepentFieldAmount = parsedAmounts[independentField]
+
+  // check if user has gone through approval process, used to show two step buttons, reset on token change
+  const [approvalSubmitted, setApprovalSubmitted] = useState<boolean>(false)
+
+  const onWallchainDrop = useCallback(() => {
+    setApprovalSubmitted(false)
+  }, [setApprovalSubmitted])
+
+  const {
+    callback: swapCallback,
+    error: swapCallbackError,
+    reason: revertReason,
+  } = useSwapCallbackV1({
+    trade,
+    deadline,
+    onWallchainDrop,
+    wallchainMasterInput,
+  })
+
+  const [{ tradeToConfirm, swapErrorMessage, attemptingTxn, txHash }, setSwapState] = useState<{
+    tradeToConfirm: SmartRouterTrade<TradeType> | undefined
+    attemptingTxn: boolean
+    swapErrorMessage: string | undefined
+    txHash: string | undefined
+  }>({
+    tradeToConfirm: undefined,
+    attemptingTxn: false,
+    swapErrorMessage: undefined,
+    txHash: undefined,
+  })
+
+  // Handlers
+  const handleConfirmDismiss = useCallback(() => {
+    setSwapState({ tradeToConfirm, attemptingTxn, swapErrorMessage, txHash })
+    // if there was a tx hash, we want to clear the input
+    if (txHash) {
+      onUserInput(Field.INPUT, '')
+    }
+  }, [attemptingTxn, onUserInput, swapErrorMessage, tradeToConfirm, txHash, setSwapState])
+
+  const handleSwap = useCallback(async () => {
+    if (
+      priceImpactWithoutFee &&
+      !confirmPriceImpactWithoutFee(
+        priceImpactWithoutFee,
+        PRICE_IMPACT_WITHOUT_FEE_CONFIRM_MIN,
+        ALLOWED_PRICE_IMPACT_HIGH,
+        t,
+      )
+    ) {
+      return undefined
+    }
+    if (!swapCallback) {
+      if (revertReason === 'insufficient allowance') {
+        setApprovalSubmitted(false)
+        setWallchainSecondaryStatus('found')
+        return undefined
+      }
+      return undefined
+    }
+    setSwapState({ attemptingTxn: true, tradeToConfirm, swapErrorMessage: undefined, txHash: undefined })
+    return swapCallback()
+      .then((res) => {
+        setWallchainSecondaryStatus('not-found')
+        setSwapState({ attemptingTxn: false, tradeToConfirm, swapErrorMessage: undefined, txHash: res.hash })
```

Comment:
Let's add a log here so we can find out how many swaps go through the previous router and like wise add one to universal router